support/util/util.go

```diff
@@ -476,3 +477,20 @@ func getHostedClusterScopeAnnotation(obj client.Object, r client.Reader) string
 	}
 	return ""
 }
+
+// SanitizeIgnitionPayload make sure the IgnitionPayload is valid
+// and does not contain inconsistencies.
+func SanitizeIgnitionPayload(payload []byte) error {
+	var jsonPayload map[string]interface{}
+
+	if err := json.Unmarshal(payload, &jsonPayload); err != nil {
+		return fmt.Errorf("error unmarshalling Ignition payload: %v", err)
+	}
+
+	_, _, err := ignitionconfig.Parse(payload)
```

Comment:
This is nice!
We could also just 
```
	var jsonPayload ignitionapi.Config

	if err := json.Unmarshal(payload, &jsonPayload); err != nil {
		return fmt.Errorf("error unmarshalling Ignition payload: %v", err)
	}
```
that way reused the existing vendored api and save the additional 14kloc of deps.
We would lose the the "duplicated entry" unit test but I think that's ok since we own this ignition creation and it's pretty much a "proxy" that redirects to the real payload via merge.source. wdyt?

Reply:
Sure, thanks for the suggestion